platform/broadcom/sonic-platform-modules-delta/ag9032v2a/modules/delta_ag9032v2a_platform.c

```diff
@@ -566,53 +566,53 @@ static ssize_t for_status(struct device *dev, struct device_attribute *dev_attr,
     mutex_lock(&dni_lock);
     switch (attr->index) {
         case SFP_IS_PRESENT:
-            /*QSFP1~8*/
-            ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_1);
-            data = (u32)(reverse_8bits(ret) & 0xff);
-            /*QSFP9~16*/
-            ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_2);
-            data |= (u32)(reverse_8bits(ret) & 0xff) << 8;
-            /*QSFP17~24*/
-            ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_3);
-            data |= (u32)(reverse_8bits(ret) & 0xff) << 16;
             /*QSFP25~32*/
             ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_4);
-            data |= (u32)(reverse_8bits(ret) & 0xff) << 24;
+            data = (u32)ret & 0xff;
+            /*QSFP17~24*/
+            ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_3);
+            data |= ((u32)ret & 0xff) << 8;
+            /*QSFP9~16*/
+            ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_2);
+            data |= (u32)(ret & 0xff) << 16;
+            /*QSFP1~8*/
+            ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_1);
+            data |= (u32)(ret & 0xff) << 24;
```

Comment:
How about reading from _1 to _4 but shift the data position first and | with return value for each reading, something like below?
data = (data<<8)|((u32)(ret&0xff))

Reply:
Attribute is show all present result like 0xFFFF from port1 to port 32.

Reply:
instead of moving ret value with different shifting value, shifting the data 8bit to the left each time before or-ing the ret value will give same shifting for each ret value. then only 8bit shifter not three different shifter (8bit, 16bit, and 24bit).
'''
data=0;
  ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_1);
data <<8;
data |= (u32)ret&0xff;
  ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_2);
data <<8;
data |= (u32)ret&0xff;
  ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_3);
data <<8;
data |= (u32)ret&0xff;
  ret = i2c_smbus_read_byte_data(pdata1[swpld1].client, QSFP_PRESENCE_4);
data <<8;
data |= (u32)ret&0xff;
'''
you can make it more simpler with macro definition too.